### DISCUSSION TITLE
自动生成摘要会显示文章限制阅读所限制的内容

### DISCUSSION THREAD

**Discussion by AR-26710** [halo-dev/halo#7012] [General]
如图所示，开启**自动生成摘要**后会显示已隐藏内容。

![Image](https://github.com/user-attachments/assets/809202d3-7cc5-4305-940c-2e683e4192b5)

算是个细节吧，因为自动生成摘要是默认开启的，所以希望能为使用文章限制阅读插件的用户做个提示，最好是为使用了阅读限制文章的自动摘要做单独处理。

**Comment by ruibaby** [comment]
2.18+ 新增了处理文章摘要的扩展点，应该可以利用这个扩展点处理一下摘要。

https://github.com/halo-dev/halo/pull/6348

cc @liuchangfitcloud

**Reply by Arch4AI** [reply]
如果按照扩展点处理，那就算文章通过验证了，他保留的还是原来的摘要？

**Reply by AR-26710** [reply]
当前是在文章设置的元数据里对文章是否启用限制阅读进行选择。

![IMG_20241126_155809.jpg](https://github.com/user-attachments/assets/6f74f668-63f1-4aae-932c-10bbf4cb7b78)

所以能否在拓展点对文章进行是否启用限制阅读功能的判断，如文章未启用则使用当前的摘要方式，反之则将摘要显示为“此篇为加密文章”（此处显示的文本可在限制阅读插件里面做一个用户可自定义文本的功能）。

**Reply by Arch4AI** [reply]
暂时没有提供这种扩展点。
